### PR TITLE
Redis limiter improvement

### PR DESCRIPTION
**Pull request by zalbiraw**
This change actually changes the algo from rolling window to leaky bucket. We will need more vetting but its a good start to improve redis performance. 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)

## Checklist

- [ ] I ensured that the documentation is up to date
- [ ] I explained why this PR updates go.mod in detail with reasoning why it's required
- [ ] I would like a code coverage CI quality gate exception and have explained why

### CODE REVIEW

File: gateway/session_manager.go [hunk 1]
```diff
@@ -54,6 +95,23 @@ func (l *SessionLimiter) doRollingWindowWrite(key, rateLimiterKey, rateLimiterSe
...
+	var batchSize int
+	var shouldSendBatch bool
+
+	if shouldBatch {
+		shouldSendBatch = l.shouldSendRedisBatch(rateLimiterKey, currentSession, globalConf, dryRun, per)
+		batchSize = globalConf.RedisRollingLimiterDRLRequestBatchSize
+
+		if batchSize < 1 {
+			batchSize = 1
+		}
```
Comment (zalbiraw) [inline gateway/session_manager.go:108]: This check should live some where else

**Comment by buger** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5772/merge
Commit:  
Triggered by: pull_request (@zalbiraw)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/6864917111)

**Comment by buger** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5772/merge
Commit: ddacb8ec7bb033f48fd22b27cfffe2a85e7bcf79 
Triggered by: pull_request (@zalbiraw)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/6869350756)

**Comment by alephnull** [comment]
Removed @TykTechnologies/devops  from reviewers as this seems to be something for the squad.

**Comment by buger** [comment]
So basically we cache failed attempts, and they do not cause load to the Redis. Thats nice! But in case if the is no rate limit hit, performance should be the same as before.

**Pull request closed without merging**